### DISCUSSION TITLE
Micropython docs for the version available in pybricks-micropython 2.0 for EV3?

### DISCUSSION THREAD

**Discussion by jaguilar** [pybricks/support#1444] [Q&A] [answered]
I was googling around for some micropython performance tips and I noticed the different code generation engines available listed [here](https://docs.micropython.org/en/latest/reference/speed_python.html#the-viper-code-emitter). It turns out that with little tweaking, Viper is able to deliver a pretty good speedup to my code -- roughly 4x. 

There's nothing about viper mentioned in the [pybricks 2.0 documentation](https://docs.pybricks.com/en/v2.0/). In fact, the whole suite of `u*` modules is missing from that page.

Now I know that EV3 is kinda a backburner project at this point, but I'm curious if there is somewhere I can go for docs on what micropython bits are available on ev3dev/pybricks-2.0. Is the [Micropython 1.11](https://docs.micropython.org/en/v1.11/index.html) documentation the best source? If it is, is there a listing of differences between Micropython 1.11 and the version of it used in pybricks? Also, has anyone had any luck getting type hints for this library installed on vscode? It's ancient so I'm guessing not but I figured I'd ask.

**Comment by dlech** [comment, marked as answer]
Yes, the official MicroPython docs should be mostly correct. The Pybricks version is basically the Unix port with most features enabled and the `pybricks` package added. So between the Pybricks 2.0 docs and the MicroPython 1.11 most things should be covered.

You can get code stubs for Pybricks 2.0 with `pip install pybricks<3` but this only contains the Pybricks package and not other MicroPython modules.

**Reply by jaguilar** [reply]
Much appreciated.
